content/en/docs/reference/kubectl/cheatsheet.md

```diff
@@ -265,6 +265,8 @@ kubectl delete pod,service baz foo                                        # Dele
 kubectl delete pods,services -l name=myLabel                              # Delete pods and services with label name=myLabel
 kubectl delete pods,services -l name=myLabel --include-uninitialized      # Delete pods and services, including uninitialized ones, with label name=myLabel
 kubectl -n my-ns delete po,svc --all                                      # Delete all pods and services, including uninitialized ones, in namespace my-ns,
+# Delete all pods matching the egrep pattern
+kubectl get pods  -n mynamespace --no-headers=true | egrep 'pattern1|pattern2'  | awk '{print  $1}' | xargs  kubectl delete -n mynamespace pod $1                                                       
```

Comment:
The Kubernetes side of this command looks fine; however, I'm thinking that you could replace `egrep` and `awk` with just `awk`.

Reply:
What role does `$1` play in this example?

Reply:
This [feature request](https://github.com/kubernetes/kubernetes/issues/72220) for JSONPath regular expression support might be relevant too.

Reply:
> What role does `$1` play in this example?

$1 is replaced with all pods names returned by egrep and awk filters.

Reply:
> The Kubernetes side of this command looks fine; however, I'm thinking that you could replace `egrep` and `awk` with just `awk`.

agreed! we can use
```kubectl get pods   --no-headers=true | awk '/pattern1|pattern2/{print $1}'  ```
instead of 
```|egrep 'pattern1|pattern2'  | awk '{print  $1}' ```